main.m

```diff
@@ -223,49 +401,63 @@ void printCertificate(NSDictionary *certificateItem) {
 }
 
-void printKey(NSDictionary *keyItem) {
-	NSString *keyClass = @"Unknown";
+void printKey(NSDictionary *keyItem)
+{
+    NSString *keyClass = @"Unknown";
 	CFTypeRef _keyClass = [keyItem objectForKey:(id)kSecAttrKeyClass];
-
-	if ([[(id)_keyClass description] isEqual:(id)kSecAttrKeyClassPublic]) {
+	int keySize = [[keyItem objectForKey:(id)kSecAttrKeySizeInBits] intValue];
+	if ([[(id)_keyClass description] isEqual:(id)kSecAttrKeyClassPublic])
+    {
 		keyClass = @"Public";
 	}
-	else if ([[(id)_keyClass description] isEqual:(id)kSecAttrKeyClassPrivate]) {
+	else if ([[(id)_keyClass description] isEqual:(id)kSecAttrKeyClassPrivate])
+    {
 		keyClass = @"Private";
 	}
-	else if ([[(id)_keyClass description] isEqual:(id)kSecAttrKeyClassSymmetric]) {
+	else if ([[(id)_keyClass description] isEqual:(id)kSecAttrKeyClassSymmetric])
+    {
 		keyClass = @"Symmetric";
 	}
-
 	printToStdOut(@"Key\n");
 	printToStdOut(@"---\n");
 	printToStdOut(@"Entitlement Group: %@\n", [keyItem objectForKey:(id)kSecAttrAccessGroup]);
 	printToStdOut(@"Label: %@\n", [keyItem objectForKey:(id)kSecAttrLabel]);
+	NSString* accessibleString = [keyItem objectForKey:(id)kSecAttrAccessible];
+	printAccessibleAttribute(accessibleString);
 	printToStdOut(@"Application Label: %@\n", [keyItem objectForKey:(id)kSecAttrApplicationLabel]);
 	printToStdOut(@"Key Class: %@\n", keyClass);
-	printToStdOut(@"Permanent Key: %@\n", CFBooleanGetValue((CFBooleanRef)[keyItem objectForKey:(id)kSecAttrIsPermanent]) == true ? @"True" : @"False");
+	if (keySize) {
+		printToStdOut(@"Permanent Key: %@\n", CFBooleanGetValue((CFBooleanRef)[keyItem objectForKey:(id)kSecAttrIsPermanent]) == true ? @"True" : @"False");
+	}
 	printToStdOut(@"Key Size: %@\n", [keyItem objectForKey:(id)kSecAttrKeySizeInBits]);
 	printToStdOut(@"Effective Key Size: %@\n", [keyItem objectForKey:(id)kSecAttrEffectiveKeySize]);
-	printToStdOut(@"For Encryption: %@\n", CFBooleanGetValue((CFBooleanRef)[keyItem objectForKey:(id)kSecAttrCanEncrypt]) == true ? @"True" : @"False");
-	printToStdOut(@"For Decryption: %@\n", CFBooleanGetValue((CFBooleanRef)[keyItem objectForKey:(id)kSecAttrCanDecrypt]) == true ? @"True" : @"False");
-	printToStdOut(@"For Key Derivation: %@\n", CFBooleanGetValue((CFBooleanRef)[keyItem objectForKey:(id)kSecAttrCanDerive]) == true ? @"True" : @"False");
-	printToStdOut(@"For Signatures: %@\n", CFBooleanGetValue((CFBooleanRef)[keyItem objectForKey:(id)kSecAttrCanSign]) == true ? @"True" : @"False");
-	printToStdOut(@"For Signature Verification: %@\n", CFBooleanGetValue((CFBooleanRef)[keyItem objectForKey:(id)kSecAttrCanVerify]) == true ? @"True" : @"False");
-	printToStdOut(@"For Key Wrapping: %@\n", CFBooleanGetValue((CFBooleanRef)[keyItem objectForKey:(id)kSecAttrCanWrap]) == true ? @"True" : @"False");
-	printToStdOut(@"For Key Unwrapping: %@\n\n", CFBooleanGetValue((CFBooleanRef)[keyItem objectForKey:(id)kSecAttrCanUnwrap]) == true ? @"True" : @"False");
-
+	if (keySize) {
```

Comment:
I can't totally tell..is there a reason we need two different `if (keySize)` conditional blocks here and above?

Reply:
Yes this is needed otherwise sometimes an error is produced which stops execution.

Reply:
I guess what I meant was, is there a reason not to just wrap all the key info bits in one `if (keySize)` block The only stuff not in there right now looks like "Key Size" and "Effective Key Size". It seems sensible to only print that info when we have `keySize`.

Reply:
I merged, the if statements.